src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -297,6 +293,11 @@ public async Task WaitForConnectionShutdownAsync()
 
             _ignoreSettingsAck = false;
             _ignoreWindowUpdates = false;
+
+            if (frame != null)
```

Comment:
Presumably you're moving this down so that we do all of the above cleanup even if frame is null.  Is it possible `await ReadFrameAsync` might throw, such that we should put the above in a finally block?  Or ReadFrameAsync won't return faulted tasks?

Reply:
yes, that was the general idea. Closing socket kicks client side even if something goes wrong and test would fail instead of hanging. I can take another look at this.